### PR TITLE
feat(tezosqa and general workflow)

### PR DESCRIPTION
**Pull request by fleandrei**
A new version 1.0 has been added to tezosqa. This version use the version 0.9.0 of smartpy. We can now set the path of smartpy-cli folder via the SMARTPY_PATH build_args argument. 
tezosqa is now added to .github/dependabot.yml. 
Workflow has slightly changed : Some Docker related parameters (the path of docker .sock socket and the connexion timeout) are now modifiable in the base_config.yml file.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -401,3 +401,28 @@ updates:
...
       - "pip"
       - "terraform"
+  # Tezosqa
+  - package-ecosystem: "docker"
+    directory: "/tezosqa/0.1"
+    schedule:
+      interval: "weekly"
+    labels:
+      - "dependencies"
+      - "pip"
```
Comment (Crow-EH) [inline .github/dependabot.yml:411]: `package-ecosystem: "docker"` so:
```suggestion
      - "docker"
```

File: tezosqa/1.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+FROM python:3.8-slim-buster
+
+ARG SMARTPY_VERSION
+ARG SMARTPY_PATH
+
+ENV SMARTPY_INSTALL_URL https://smartpy.io/releases/${SMARTPY_VERSION}/cli
+
+RUN echo "Install system dependencies for python and pip" && \
+    #echo "path : " && smpat = $(${SMARTPY_PATH}) && mkdir ${smpat} && ls -la ~ && exit 1 &&\
```
Comment (Crow-EH) [inline tezosqa/1.0/Dockerfile:9]: dead code

File: tezosqa/1.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    echo "Done!" && \
+    echo "Install crypto libs" && \
+    apt-get update -q -y && apt-get install -q -y libsodium-dev libsecp256k1-dev libgmp-dev gcc git g++ make && \
+    echo "Done!" && \
+    echo "Install SmartPyDev" && \
+    apt-get install -q -y curl && \
+    curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
+    apt-get install -y nodejs && \
+    curl -s "${SMARTPY_INSTALL_URL}/install.sh" > install.sh && bash install.sh --prefix ${SMARTPY_PATH} --from ${SMARTPY_INSTALL_URL} --yes && rm install.sh && \
+    #ls -la / && ls -la ${SMARTPY_PATH} && exit 1 && \
```
Comment (Crow-EH) [inline tezosqa/1.0/Dockerfile:23]: dead code

File: tezosqa/1.0/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    pip install pipenv && \
+    echo "Done!" && \
+    echo "Install crypto libs" && \
+    apt-get update -q -y && apt-get install -q -y libsodium-dev libsecp256k1-dev libgmp-dev gcc git g++ make && \
+    echo "Done!" && \
+    echo "Install SmartPyDev" && \
+    apt-get install -q -y curl && \
+    curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
+    apt-get install -y nodejs && \
+    curl -s "${SMARTPY_INSTALL_URL}/install.sh" > install.sh && bash install.sh --prefix ${SMARTPY_PATH} --from ${SMARTPY_INSTALL_URL} --yes && rm install.sh && \
```
Comment (Crow-EH) [inline tezosqa/1.0/Dockerfile:22]: You don't need to rm anything if you don't create a file in the first place, check if this works:
```suggestion
     bash <(curl -s "${SMARTPY_INSTALL_URL}/install.sh") --prefix ${SMARTPY_PATH} --from ${SMARTPY_INSTALL_URL} --yes && \
```

File: .github/dependabot.yml [hunk 5]
```diff
@@ -401,3 +401,28 @@ updates:
...
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major", "version-update:semver-minor"]
+  - package-ecosystem: "docker"
+    directory: "/tezosqa/1.0"
+    schedule:
+      interval: "weekly"
+    labels:
+      - "dependencies"
+      - "pip"
```
Comment (Crow-EH) [inline .github/dependabot.yml:422]: docker

File: .github/dependabot.yml [hunk 6]
```diff
@@ -401,3 +401,28 @@ updates:
...
+      - "tezosqa"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major", "version-update:semver-minor"]
+
+
```
Comment (Crow-EH) [inline .github/dependabot.yml:428]: There should be only one (empty) newline at the end.

File: base_config.yml [hunk 7]
```diff
@@ -1,8 +1,11 @@
 ---
 docker_hub_namespace: &docker_hub_namespace ekino
+DOCKER_SOCK_PATH: &DOCKER_SOCK_PATH unix://var/run/docker.sock
```
Comment (Crow-EH) [inline base_config.yml:3]: you don't need to declare an anchor (`&DOCKER_SOCK_PATH`) if you don't use the reference anywhere else in a yaml file (`*DOCKER_SOCK_PATH`).

File: base_config.yml [hunk 8]
```diff
@@ -1,8 +1,11 @@
...
+DOCKER_TIMEOUT: &DOCKER_TIMEOUT 600
```
Comment (Crow-EH) [inline base_config.yml:4]: Same thing, the anchor isn't used anywhere

**Review by Crow-EH — CHANGES_REQUESTED** [review]
Thank you for you contribution ! Here are some aditionnal comments :

# The root Dockerfile isn't used anymore
You shouldn't need the root Dockerfile anymore since you made a directory for each version, so please delete `tezosqa/Dockerfile`.

# DRY
It seems like, excluding dead code, only smartpy install is different: That's an oportunity to not repeat yourself. 😄 

You should consider creating a common script in the parent directory with a simple `if`. Look at PHP images  and their `run.sh` for reference.

You just need to `COPY run.sh` as if it was in the same directory as your Dockerfile, we set up the the parent directory as build context for this use case. 🙂

# Wording and changelog
* Please respect this pattern for your commit `<type>(<optional scope>): <description>`. Suggestion: `feat(tezosqua): add version 1.0 and expose docker variable`.
* Please update `CHANGELOG.md`. Just add a new section at the start with a partial date `2022-03`

File: base_config.yml [hunk 9]
```diff
@@ -1,8 +1,11 @@
 ---
 docker_hub_namespace: &docker_hub_namespace ekino
```
Comment (Crow-EH) [inline base_config.yml:2]: Remove this anchor too while you're at it :p

**Review by Bladrak — CHANGES_REQUESTED** [review]

File: base_config.yml [hunk 10]
```diff
@@ -1,8 +1,12 @@
 ---
-docker_hub_namespace: &docker_hub_namespace ekino
+docker_hub_namespace: ekino
+#DOCKER_SOCK_PATH: unix://var/run/docker.sock
```
Comment (Bladrak) [inline base_config.yml:3]: Remove the commented line if it's not used.

File: .github/dependabot.yml [hunk 11]
```diff
@@ -401,3 +401,26 @@ updates:
       - "dependencies"
       - "pip"
       - "terraform"
+  # Tezosqa
+  - package-ecosystem: "docker"
```
Comment (Bladrak) [inline .github/dependabot.yml:405]: We don't need to keep 2 versions, remove the 0.1 one and keep the 1.0 one.

**Review by Bladrak — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 11 above)
Comment (Crow-EH) [reply .github/dependabot.yml:405]: ah then the script and the subdirectories are useless and you can go back to a single root Dockerfile if you want @fleandrei , sorry. Let's say you now know how to it. 😄

File: base_config.yml [hunk 12]
```diff
@@ -1,8 +1,12 @@
...
+DOCKER_SOCK_PATH: unix://Users/andrei.fleiser/docker.sock
```
[the remark this answers is not in the document]
Comment (Crow-EH) [reply base_config.yml:4]: I'm not entirely sure that would be `unix://var/run/docker.sock` on the CI (and it needs to), but I guess you can test.

Othewise just stick to `unix://var/run/docker.sock` or, if you want to and have some time, modify the python script to use the `DOCKER_HOST` env variable (should be `unix://Users/andrei.fleiser/docker.sock` on your device) if it exists, or `DOCKER_SOCK_PATH` if it doesn't.

File: tezosqa/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    && echo "Install Pytezos" \
+    && apt-get install -q -y pkg-config \
+    && pip install pytezos pytest ipython fire pysodium secp256k1 'fastecdsa<2.0.0' \
+    && echo "Done!" \
+    && echo "Cleanup" \
+    && apt-get -q -y autoremove \
+    && apt-get -q -y clean && apt-get -q -y purge \
+    && rm -rf /var/lib/apt/lists/* /var/lib/dpkg/*-old \
+    && echo "Done!" 
+
```
Comment (Bladrak) [inline tezosqa/Dockerfile:38]: Extra blank line to remove

**Review by Bladrak — CHANGES_REQUESTED** [review]

**Review by Bladrak — APPROVED** [review]

**Comment by Bladrak** [comment]
Some conflicts to fix, you should rebase it. And maybe squash it so there's only one commit :)

**Review by Bladrak — APPROVED** [review]

**Pull request merged**